src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -202,6 +202,13 @@ private static Expression[] CreateArguments(ParameterInfo[]? parameters, Factory
             var errorMessage = BuildErrorMessageForInferredBodyParameter(factoryContext);
             throw new InvalidOperationException(errorMessage);
         }
+
+        if (factoryContext.HasInferredBody && factoryContext.HasInferredQueryArray)
+        {
+            var errorMessage = BuildErrorMessageForQueryAndJsonBodyParameters(factoryContext);
+            throw new InvalidOperationException(errorMessage);
```

Comment:
Is this currently possible considering `factoryContext.DisableInferredFromBody` must be true for `factoryContext.HasInferredBody` to be set? What if `[FromQuery]` is explicitly used?

Reply:
I think you're right, I would need to change the order to get a "better exception" but maybe it isn't worth it?